src/input.service.ts

```diff
@@ -86,7 +86,7 @@ export class InputService {
         let {allowNegative, decimal, precision, prefix, suffix, thousands, min, max, inputMode} = this.options;
 
         rawValue = isNumber ? new Number(rawValue).toFixed(precision) : rawValue;
-        let onlyNumbers = rawValue.replace(this.ONLY_NUMBERS_REGEX, "");
+        let onlyNumbers = rawValue.replace(suffix,"").replace(this.ONLY_NUMBERS_REGEX, "");
```

Comment:
I think that `suffix` can be anything and it cannot be simply removed from any position of `rawValue`. It would be better to check that the string ends with `suffix` and remove `suffix.length` characters from the end of `rawValue`.
I think you can similarly remove `prefix`.